### PR TITLE
Build: improve CI, add licence header checks, clean up example apps

### PR DESCRIPTION
**Pull request by vadymmarkov**
- Remove `Danger` since it created too much unwanted noise in pull requests
- Remove `CocoaPods` example app in favour of `VideoFilters` app that also uses `CocoaPods` to install the SDK
- Add new GitHub workflow to build and test both the SDK and example apps
- Add licence header checks
- Update SwiftLint version and fix new warnings
- Add Git pre-commit hook that can be used to automatically update license headers and run SwiftLint

### CODE REVIEW

File: Scripts/license_headers.sh [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        echo "$(echo -e "$header\n"; cat $file)" > $file
+        echo "Added license header in $file"
+      else
+        echo "::error::License header is missing in $file"
+        ((errors++))
+      fi
+    fi
+    if ! grep -q "$current_year" "$file"; then
+      if [[ $update -eq 1 ]]; then
+        str="s/Copyright.*Pexip/Copyright 2022-$current_year Pexip/"
```
Comment (tpettersen) [inline Scripts/license_headers.sh:37]: Is hardcoded 2022 here correct? If you add a new file in 2023, should it say Copyright 2022-2023 then?

**Review by tpettersen — APPROVED** [review]
Looks good :) I had a small comment in your script

File: Scripts/license_headers.sh (hunk 1 above)
Comment (vadymmarkov) [reply Scripts/license_headers.sh]: You're right, if the file was created in 2023 it makes more sense to say Copyright 2023; or Copyright 2023-2024 if it gets updated next year. Should be fixed now.

**Review by tpettersen — APPROVED** [review]

File: Scripts/license_headers.sh (hunk 1 above)
Comment (tpettersen) [reply Scripts/license_headers.sh]: Looks good now :)

**Pull request merged**